### PR TITLE
Bug 1832722: remove `project-name` query param when active namespace is changed

### PR DESCRIPTION
**Pull request by debsmita1**
**Fixes**:
https://issues.redhat.com/browse/ODC-3795

**Root Cause/Analysis**:
When a project is searched from the list of all projects and a particular project is selected, the url is updated with the current namespace but the searched project filter persists as query param in the URL

**Solution Description**:
When a project is selected from the all-projects list, the `project-name` query param is removed from the URL

**GIF**:
n1

/kind bug

### CODE REVIEW

**Comment by christianvogt** [comment]
@debsmita1 this negatively impacts every resource list page in admin console where switching the namespace now will remove their filter. I don't believe we can do such a change.

I thought you were going to remove the query param from the table links so at least when the user clicks on those links after adding a project name filter, it would work.

**Comment by debsmita1** [comment]
> @debsmita1 this negatively impacts every resource list page in admin console where switching the namespace now will remove their filter. I don't believe we can do such a change.
> 
> I thought you were going to remove the query param from the table links so at least when the user clicks on those links after adding a project name filter, it would work.

updated my PR, now removing only the `project-name` filter when a project is selected from the table links.

File: frontend/public/components/namespace.jsx [hunk 1]
```diff
@@ -275,7 +278,13 @@ const ProjectLink = connect(null, { setActiveNamespace: UIActions.setActiveNames
         title={project.metadata.name}
         type="button"
         className="co-resource-item__resource-name"
-        onClick={() => setActiveNamespace(project.metadata.name)}
+        onClick={() => {
+          setActiveNamespace(project.metadata.name);
+          removeQueryArgument('project-name');
+          store.dispatch(
+            k8sActions.filterList(referenceForModel(ProjectModel), 'project-name', ''),
+          );
```
Comment (christianvogt) [inline frontend/public/components/namespace.jsx:286]: Why is this necessary? 

Also, `store` is accessed globally instead of through `connect`,

Comment (debsmita1) [reply frontend/public/components/namespace.jsx:286]: @christianvogt When I search and select a project from the table links and then go back to `all-projects` from the namespace dropdown, the URL has no sign of project-name` query param, also the input filter is empty but the ListPageWrapper_ component receives filters: {project-name : <abc>} from the firehose. So this code change takes care of that, I am unsure if the bug fix for https://issues.redhat.com/browse/ODC-3896 should take care of this. Let me know your views

File: frontend/public/components/namespace.jsx [hunk 2]
```diff
@@ -662,11 +671,14 @@ class NamespaceBarDropdowns_ extends React.Component {
       if (newNamespace === CREATE_NEW_RESOURCE) {
         createProjectModal({
           blocking: true,
-          onSubmit: (newProject) => setActiveNamespace(newProject.metadata.name),
+          onSubmit: (newProject) => {
+            setActiveNamespace(newProject.metadata.name), removeQueryArgument('project-name');
```
Comment (christianvogt) [inline frontend/public/components/namespace.jsx:675]:
```suggestion
            setActiveNamespace(newProject.metadata.name);
            removeQueryArgument('project-name');
```

File: frontend/public/components/namespace.jsx (hunk 1 above)
Comment (christianvogt) [reply frontend/public/components/namespace.jsx:286]: I feel like this is a patch for the real problem lurking with https://issues.redhat.com/browse/ODC-3896
Maybe if we keep it at least we need a comment.

Regardless we need to avoid using the redux store as a global and instead use connect.
